File: backend/app/models/resources.py
from __future__ import annotations

from typing import Any, Dict, List, Optional

from pydantic import BaseModel


class Resource(BaseModel):
    id: str
    title: str
    source_url: str
    source_type: str
    content: str
    summary: str
    last_fetched: Optional[str] = None
    linked_ticket_id: Optional[str] = None


class ResourceSummary(BaseModel):
    title: str
    source_url: str
    content: str


class ResourceFetchRequest(BaseModel):
    urls: List[str]
    source_type: str = "url"
    max_content_length: int = 4000


class ResourceFetchResponse(BaseModel):
    resources: List[Resource]
    errors: Dict[str, str]

File: frontend/src/types/resources.ts
export interface Resource {
  id: string;
  title: string;
  source_url: string;
  source_type: string;
  content: string;
  summary: string;
  last_fetched: string | null;
  linked_ticket_id: string | null;
}

export interface ResourceFetchRequest {
  urls: string[];
  source_type?: string;
  max_content_length?: number;
}

export interface ResourceFetchResponse {
  resources: Resource[];
  errors: Record<string, string>;
}
